src/app/application-dialog/application-dialog.component.ts

```diff
@@ -97,12 +97,41 @@ export class ApplicationDialogComponent implements OnInit {
   onDoneClick(): void {
     //form validation and post to backend
     this.saveLoanDetails();
-    console.log(`The dialog was closed with result: ${this.applicationData.realEstatePrice}`);
+    console.log(this.applicationData)
+    this.api.postApplication(this.applicationData).subscribe({
+      next: (success) => {
+        console.log(success);
+      },
+      error: (err) => {
+        console.log(err);
+      }
+    })
   }
 
   saveLoanDetails(): void {
     this.applicationData.realEstatePrice = this.loanDetailsForm.value.realEstatePrice;
-//update other fields
+    this.applicationData.downPayment = this.loanDetailsForm.value.downPayment;
```

Comment:
Looks repetitive, update to
[‘key1’, ‘key2’, …].forEach((key) => { this.applicationData[key] = this.loanDetailsForm.value[key]});
for style points 😎

Reply:
😎😎😎😎 agreed 😎😎😎😎

Reply:
Bindings after forEach loop are no longer needed
Just make sure form data sent is valid